src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.time.LocalDate;
+import java.util.List;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.service.MovieSessionService;
+
+@Service
+public class MovieSessionServiceImpl implements MovieSessionService {
+    @Inject
+    private MovieSessionDao movieSessionDao;
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        return movieSessionDao.add(movieSession);
+    }
+
+    @Override
+    public MovieSession get(Long id) {
+        return movieSessionDao.get(id)
+                .orElseThrow(() -> new EntityNotFoundException("Movie session with id = ["
+                        + id + "] not found"));
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        return movieSessionDao.findAvailableSessions(movieId, date);
```

Comment:
Ensure that the `MovieSessionDao.findAvailableSessions` method is implemented in such a way that it only retrieves movie sessions within the provided date's day and does not require additional filtering on the service layer. This is important for performance reasons as it's better to extract 'ready to use' data directly from the database on the DAO layer. If the current implementation retrieves all sessions and filters them on the service layer, it should be refactored to filter on the DAO layer.